### ISSUE TITLE
Game preformance issue

### ISSUE THREAD

**Issue by Jninja1** [CleverRaven/Cataclysm-DDA#57152]
### Describe the bug

The game begins to notably slow down once four or more solar panels are connect to each other with an extension cord of sorts is connected to one leading to a battery.

### Steps to reproduce

1) Install solar panels as an appliance
2) Install any sort of battery as an appliance
3) Connect panels to battery or vice versa 

### Expected behavior

I expected the game to run with no noticeable slow down and provide power to the batteries.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22000.613 (21H2)
- Game Version: 44c51a4 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Cataclysm++ [Cata++],
    Locked and Loaded [Locked_and_Loaded],
    MADA Mod [MADA_bows_rebalance],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills]
]

**Comment by Inglonias** [comment]
can you provide the save file where this is happening, or a screenshot explaining your power setup?

**Comment by robob27** [comment]
I can provide a save if needed but it's very easy to repro. I tested how long it took my PC to wait 6 hours before and after making this setup (linear, one connection between each device):

![image](https://user-images.githubusercontent.com/2104555/186037211-887894f2-fd89-4119-9b4c-cbbe5325a2ce.png)

Before setup: 11.32s
After setup: 17.83s

I made a slightly bigger grid than the creator of the issue described. This is because when I tested the grid setup described by the creator, there was only about a 1 second difference on my hardware (AMD Ryzen 7 3700X @ 3.60 GHz) but I imagine even their setup is noticeably slower on older hardware. It scales very quickly, adding only a few more solar panels and an oven and I'm consistently waiting about 6 seconds longer for sleeping.. and this is still a very simple setup.

I first noticed this issue myself the other day. After setting up ~15 solar panels, a battery and several appliances, it all of a sudden took _forever_ to sleep/craft/read at my base (this was on slower hardware, a 2.30GHz cpu). I checked github and saw there had previously been some issues with setups with parallel connections, or many redundant connections, so I redid the setup to make sure there was none of that, but it was still just as slow.
